zkevm-circuits/src/tx_circuit/test.rs

```diff
@@ -146,7 +146,12 @@ fn build_eip1559_tx(id: usize) -> Transaction {
     tx.max_priority_fee_per_gas = eth_tx.max_priority_fee_per_gas.unwrap_or(U256::zero());
     tx.call_data = eth_tx.input.to_vec();
     tx.callee_address = eth_tx.to;
-    tx.caller_address = eth_tx.from;
+
+    tx.caller_address = H160::from_slice(&[
+        0x20, 0x2b, 0xB2, 0xfa, 0xB1, 0xe3, 0x5D, 0x94, 0x0F, 0xde, 0x99, 0xb2, 0x14, 0xba, 0x49,
+        0xDA, 0xfb, 0xce, 0xf6, 0x2A,
+    ]);
```

Comment:
@darth-cy what happened here?

Reply:
The invocation on line 127 `EthTransaction::decode` failed to decode rlp bytes correctly and always produces a tx with caller address `0x000...000`. The resulting 1559 tx always gets treated as padding as the caller address is 0x0. Since this is a method in a dependency, the only option is to manually assign the correct caller address.

Reply:
Cc @DreamWuGit similar problem here. Maybe we have issues with signing 1559

Reply:
@darth-cy https://github.com/scroll-tech/zkevm-circuits/pull/1182 merged, you can have a look whether you still need this adhoc change. If yes,  add a "FIXME" comment.